packages/manager/modules/iam/src/dashboard/dashboard.routing.js

```diff
@@ -2,8 +2,24 @@ export default /* @ngInject */ ($stateProvider) => {
   $stateProvider.state('iam.dashboard', {
     url: '/dashboard',
     component: 'iamDashboard',
-    redirectTo: 'iam.dashboard.policies',
+    redirectTo: (transition) =>
+      transition
+        .injector()
+        .getAsync('iam')
+        .then(({ authorized }) =>
+          authorized ? { state: 'iam.dashboard.policies' } : false,
+        ),
     resolve: {
+      iam: /* @ngInject */ (IAMService) =>
+        IAMService.getPolicies({})
+          .then(() => ({
+            authorized: true,
+            details: null,
+          }))
+          .catch(({ data }) => ({
+            authorized: false,
+            details: data?.details?.unauthorizedActionsByIAM,
+          })),
```

Comment:
**nitpick:** Error can be because of API unavailability in which case we have to display the Error page and not the unauthorised message.
Display the message only when the response is 403.

Also, I dont like the idea of changing the signature of the method to Object. May be you can return `null` and use that to differentiate.